components/batch-system/src/router.rs

```diff
@@ -118,16 +124,45 @@ where
     where
         F: FnMut(&BasicMailbox<N>) -> Option<R>,
     {
+        let maybe_clean_cache_leak = |alive_cnt, caches: &mut LruCache<u64, BasicMailbox<N>>| {
+            let leak = self.leak_count(alive_cnt);
+            for _ in 0..leak {
+                // Maybe the cache is hold leaked mailbox, so we need to remove it.
+                let _ = caches.remove_lru();
```

Comment:
Or how about using lock-free hashmap directly, like `DashMap`, the read-heavy performance seems good enough https://github.com/xacrimon/conc-map-bench#ready-heavy-fx-hasher

We can run a benchbot to see how much performance it affects. If nearly zero, we can use it to simplify code

Reply:
Or how about adding a timestamp for cache items, and clean inactive items periodically

Reply:
> Or how about using lock-free hashmap directly, like DashMap ...

Good suggestion! Though I prefer to make a minimum fix in this PR.

> how about adding a timestamp for cache items, and clean inactive items periodically

Compared to the get operation in HashMap, retrieving timestamps may take more time and could potentially result in performance issues.